pkg/kubectl/cmd/util/factory_object_mapping.go

```diff
@@ -162,14 +158,61 @@ func (f *ring1Factory) Describer(mapping *meta.RESTMapping) (kubectl.Describer,
 			return &kubectl.ClusterDescriber{Interface: fedClientSet}, nil
 		}
 	}
+
 	clientset, err := f.clientAccessFactory.ClientSetForVersion(&mappingVersion)
 	if err != nil {
+		// if we can't make a client for this group/version, go generic if possible
+		if genericDescriber, genericErr := f.genericDescriber(mapping); genericErr == nil {
```

Comment:
Explain why this peer call is safe. Peer calls are normally banned, but this peer call is legal because its not logically a separate function, its just a big macro.  If anyone ever exposes it, then the peer call becomes illegal.